### PR TITLE
CLN: FrameOrSeries(Union)

### LINKED ISSUE #255: CLEAN:  Replace FrameOrSeries TypeVar with NDFRameT, rename FrameOrSeriesUnion to be FrameOrSeries
We currently have this in `_typing.pyi`:
```
FrameOrSeries = TypeVar("FrameOrSeries", bound=NDFrame)
FrameOrSeriesUnion = Union[DataFrame, Series]
NDFrameT = TypeVar("NDFrameT", bound=NDFrame)
```

`FrameOrSeries` and `NDFrameT`  clearly are duplicates.

We should remove `FrameOrSeries` from the code and replace it with `NDFrameT`

If we do that, then we can also replace `FrameOrSeriesUnion` and call it `FrameOrSeries`

### PR DESCRIPTION
**Pull request by twoertwein**
- [x] Closes #255

Pandas does not define `FrameOrSeriesUnion` and instead uses `DataFrame | Series`.

### CODE REVIEW

**Comment by bashtage** [comment]
Maybe move nightly to be first?  Sort of a pain to know that only nightly failed.

**Comment by twoertwein** [comment]
Honestly, I wouldn't mind adding continue-on-error until nightly passes. Once it passes, we can remove it until it errors again and then either fix it/ignore it temporarily.

File: pandas-stubs/_typing.pyi [hunk 1]
```diff
@@ -80,14 +79,10 @@ class WriteExcelBuffer(WriteBuffer[bytes], Protocol):
...
 
 FilePath = Union[str, PathLike[str]]
+FilePathOrBuffer = Union[
+    FilePath, IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper, mmap
+]
 
-Buffer = Union[IO[AnyStr], RawIOBase, BufferedIOBase, TextIOBase, TextIOWrapper, mmap]
-FileOrBuffer = Union[str, Buffer[AnyStr]]
-FilePathOrBuffer = Union[PathLike[str], FileOrBuffer[AnyStr]]
-FilePathOrBytesBuffer = Union[PathLike[str], WriteBuffer[bytes]]
```
Comment (twoertwein) [inline pandas-stubs/_typing.pyi:87]: All these buffers do not exist in pandas. Only `FilePathOrBytesBuffer` is used in the stubs, so I merged them.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,6 +3,33 @@ name: 'Test'
 on:
   [push, pull_request, workflow_dispatch]
 jobs:
+
+  nightly:
```
Comment (Dr-Irv) [inline .github/workflows/test.yml:7]: should the change in `test.yml` be in a separate PR?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (twoertwein) [reply .github/workflows/test.yml]: removed

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein
